lib/questions.rb

```diff
@@ -1,226 +1,292 @@
-# should return 20
-def get_upper_limit_of(range)
-end
-
-# should return true for a 3 dot range like 1...20, false for a 
-# normal 2 dot range
-def is_a_3_dot_range?(range)
-end
-
-# get the square root of a number
-def square_root_of(number)
-end
-
-# count the number of words in a file
-def word_count_a_file(file_path)
-end
-
-# --- tougher ones ---
-
-# call an arbitrary method from a string. so if I
-# called call_method_from_string('foobar')
-# the method foobar should be invoked
-def call_method_from_string(str_method)
-end
-
-# return true if the date is a uk bank holiday for 2014
-# the list of bank holidays is here:
-# https://www.gov.uk/bank-holidays
-def is_a_2014_bank_holiday?(date)
-end
-
-# given your birthday this year, this method tells you
-# the next year when your birthday will fall on a friday
-# e.g. january 1st, will next be a friday in 2016
-# return the day as a capitalized string like 'Friday'
-def your_birthday_is_on_a_friday_in_the_year(birthday)
-end
-
-# in a file, total the number of times words of different lengths
-# appear. So in a file with the text "the cat sat on the blue mat"
-# I have 5 words which are 3 letters long, 1 which is 2 letters long
-# and 1 that is 4 letters long. Return it as a hash in the format
-# word_length => count, e.g. {2 => 1, 3 => 5, 4 => 1}
-def count_words_of_each_length_in_a_file(file_path)
-end
-
-# implement fizzbuzz without modulo, i.e. the % method
-# go from 1 to 100
-# (there's no RSpec test for this one)
-def fizzbuzz_without_modulo
-end
-
-# print the lyrics of the song 99 bottles of beer on the wall
-# http://www.99-bottles-of-beer.net/lyrics.html
-# make sure you use the singular when you have one bottle of 
-# beer on the wall, and print 'no more bottles of beer on the wall'
-# at the end.
-# (there's no RSpec test for this one)
-def ninety_nine_bottles_of_beer
-end
+  # keep only the elements that start with an a
+  def select_elements_starting_with_a(array)
+    array.keep_if {|arr| arr.start_with?'a'}
+  end
+
+  # keep only the elements that start with a vowel
+  def select_elements_starting_with_vowel(array)
+    array.keep_if {|arr| arr.start_with?'a'}
+  end
+
+  # remove instances of nil (but NOT false) from an array
+  def remove_nils_from_array(array)
+    array.compact
+  end
+
+  # remove instances of nil AND false from an array
+  def remove_nils_and_false_from_array(array)
+    array.keep_if { |v| v =~ /[abc]/ }
+  end
+
+  # don't reverse the array, but reverse every word inside it. e.g.
+  # ['dog', 'monkey'] becomes ['god', 'yeknom']
+  def reverse_every_element_in_array(array)
+    array.each {|word| word.reverse!}
+  end
+
+  # given an array of student names, like ['Bob', 'Dave', 'Clive']
+  # give every possible pairing - in this case:
+  # [['Bob', 'Clive'], ['Bob', 'Dave'], ['Clive', 'Dave']]
+  # make sure you don't have the same pairing twice,
+  def every_possible_pairing_of_students(array)
```

Comment:
Unused method argument - `array`. If it's necessary, use `_` or `_array` as an argument name to indicate that it won't be used. You can also write as `every_possible_pairing_of_students(*)` if you want the method to accept any arguments but don't care about them.